### PR TITLE
Code cleanup and fix for google maps api

### PR DESCRIPTION
**Pull request by styfle**
This is a lot of code cleanup and a fix for the google maps api.
I removed a lot of code that seemed unnecessary.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,27 +30,29 @@ app.get('/', function(request, response) {
   response.render('pages/index')
 });
 
-// When bot is called, reply with link to app '<app_url>/geoloc.htm'
+app.get('/findme', function(request, response) {
```
Comment (styfle) [inline index.js]: Note `/geoloc.htm` is now `/findme`

File: package.json [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
+    "ejs": "^2.5.6",
+    "express": "^4.15.3"
   },
   "engines": {
-    "node": "0.12.2"
+    "node": "8.1.4"
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/concurlabs/geoslack"
```
Comment (styfle) [inline package.json]: @ismaelc I reverted the repo url.

File: public/js/h5utils.js [hunk 3]
```diff
@@ -1,65 +0,0 @@
-// For discussion and comments, see: http://remysharp.com/2009/01/07/html5-enabling-script/
```
Comment (styfle) [inline public/js/h5utils.js]: I don't think this file was used

File: views/pages/findme.ejs [hunk 4]
```diff
@@ -1,26 +1,22 @@
...
     </header>
-<meta name="viewport" content="width=620" />
-
-<script type="text/javascript" src="https://maps.google.com/maps/api/js?sensor=false"></script>
     <article>
       <p>Finding your location: <span id="status">checking...</span></p>
     </article>
-<script>
+
+<script type="text/javascript" src="https://maps.google.com/maps/api/js?key=<%=process.env.GMAPS_API_KEY%>"></script>
```
Comment (styfle) [inline views/pages/findme.ejs]: Here is where `GMAPS_API_KEY` gets used.

File: views/pages/findme.ejs [hunk 5]
```diff
@@ -55,28 +51,25 @@
...
-  $.post("/coords", { lat: position.coords.latitude, lng: position.coords.longitude, user: user }).done(function( data ){
-    window.open('', '_self', '');
-  	window.close();
-  });
+  if (!localStorage.getItem('LocateMe')) {
+    localStorage.setItem('LocateMe', randomString(5));
+  }
+  const user = localStorage.getItem('LocateMe');
+  const data = { lat: position.coords.latitude, lng: position.coords.longitude, user: user };
+  fetch('/coords', {
```
Comment (styfle) [inline views/pages/findme.ejs]: I am taking advantage of `fetch` instead of using jquery.ajax. Most modern browsers support it. https://caniuse.com/#search=fetch

File: views/pages/findme.ejs [hunk 6]
```diff
@@ -55,28 +51,25 @@
...
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(data)
+  }).then(res => {
+    //window.open('', '_self', '');
```
Comment (styfle) [inline views/pages/findme.ejs]: I don't really know the purpose of this window.open and window.close so I removed it

File: views/pages/findme.ejs [hunk 7]
```diff
@@ -91,7 +84,6 @@
 
 </script>
 </section>
-<script src="js/prettify.packed.js"></script>
```
Comment (styfle) [inline views/pages/findme.ejs]: I don't know the purpose of this prettify code so I removed it

File: package.json [hunk 8]
```diff
@@ -1,28 +1,27 @@
...
-    "express": "~4.9.x",
-    "body-parser": "1.13.2",
-    "querystring": "0.2.0"
+    "body-parser": "^1.17.2",
+    "ejs": "^2.5.6",
+    "express": "^4.15.3"
   },
   "engines": {
-    "node": "0.12.2"
+    "node": "8.1.4"
```
Comment (styfle) [inline package.json]: Node.js has come a long way 😄

File: index.js [hunk 9]
```diff
@@ -33,27 +30,29 @@ app.get('/', function(request, response) {
...
 var people = [];
 var pplCtr = 0;
 
-// Called by index.ejs (geoloc.htm). Receives the coordinates from HTML5 geolocation
+// Called by front-end. Receives the coordinates from HTML5 geolocation
 app.post('/coords', function(request, response) {
-
-	var latlng = request.body.lat + "," + request.body.lng;
-	var user = request.body.user;
+	const { body, user, lat, lng } = request;
```
Comment (styfle) [inline index.js]: Using ES6 destructuring because its 2017 😄

File: index.js [hunk 10]
```diff
@@ -13,8 +10,8 @@ var querystring = require('querystring');
 var https = require('https');
 var config = require('./config');
 
-app.use(bodyParser.urlencoded({ extended: false }));
 app.set('port', (process.env.PORT || 5000));
+app.use(bodyParser.json());
```
Comment (styfle) [inline index.js]: The body we expect is json from the `fetch` call.
